configuration/pih/migrations/tb_prophylaxis_order_reason.sql

```diff
@@ -0,0 +1,28 @@
+-- the following selects all Isoniazide orders 
+-- except where the order coincides (using end date as the identifying period) 
+-- with at least one of the 3 drugs used in cocktails with it
+-- these are updated with an order reason = "TB Prophylaxis"
+
+drop temporary table if exists temp_tb_prophylaxis_orders;
+create temporary table temp_tb_prophylaxis_orders
+select o.order_id 
+from orders o
+where o.concept_id = concept_from_mapping('PIH','ISONIAZID')
+and o.order_action = 'NEW' 
+and o.voided = 0
+and not EXISTS 
+	(select 1 from orders o2
+	where o2.patient_id = o.patient_id
+	and o2.voided = 0
+	and o2.date_activated <= COALESCE(o.date_stopped,auto_expire_date)
+	and o2.date_stopped >= COALESCE(o.date_stopped,auto_expire_date)
+	and o2.order_action = 'NEW' 
+	and o2.concept_id IN 
```

Comment:
Rather than limiting to these specific concepts (which won't cover everything, particularly any multi-ingredient combination drug), it might be better to limit to any order that has order_reason = 'TB Treatment' and which is not the same as the INH order that is being checked.